### PR TITLE
Updates required for Rails 6.1

### PR DESCRIPTION
**Pull request by vivekbisen**
#57

### CODE REVIEW

File: lib/active_record/connection_adapters/percona_adapter.rb [hunk 1]
```diff
@@ -74,6 +74,12 @@ def initialize(connection, _logger, connection_options, _config)
         @prepared_statements = false
       end
 
+      def write_query?(sql) # :nodoc:
```
Comment (vivekbisen) [inline lib/active_record/connection_adapters/percona_adapter.rb:77]: Needs to be [implemented](https://github.com/rails/rails/blob/6-1-stable/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb#L112). Brought in from [mysql](https://github.com/rails/rails/blob/6-1-stable/activerecord/lib/active_record/connection_adapters/mysql/database_statements.rb#L22-L29).

**Comment by wyhaines** [comment]
:eyes:

**Comment by vivekbisen** [comment]
Still got some work to do on this. Once the PR is ready, I'll change this draft PR to Ready for Review.

File: lib/active_record/connection_adapters/percona_adapter.rb [hunk 2]
```diff
@@ -119,16 +128,28 @@ def new_column(field, default, type_metadata, null, table_name, default_function
...
-        index_name, index_type, index_columns, index_options = add_index_options(table_name, column_name, options)
-        execute "ALTER TABLE #{quote_table_name(table_name)} ADD #{index_type} INDEX #{quote_column_name(index_name)} (#{index_columns})#{index_options}" # rubocop:disable Metrics/LineLength
+        if ActiveRecord::VERSION::STRING >= '6.1'
+          index, algorithm, if_not_exists = add_index_options(table_name, column_name, options)
+          create_index = CreateIndexDefinition.new(index, algorithm, if_not_exists)
+          execute schema_creation.accept(create_index)
+        else
+          index_name, index_type, index_columns, index_options = add_index_options(table_name, column_name, options)
+          execute "ALTER TABLE #{quote_table_name(table_name)} ADD #{index_type} INDEX #{quote_column_name(index_name)} (#{index_columns})#{index_options}" # rubocop:disable Metrics/LineLength
+        end
```
Comment (vivekbisen) [inline lib/active_record/connection_adapters/percona_adapter.rb:138]: This generates/executes different sql tested [here](https://github.com/departurerb/departure/pull/58/files#diff-c150ab33c3d9469f6de8620566924b97e6b089ee95dff7743c733b5fdbebedf7R125). Should it keep the sql same? Not sure if `ALTER TABLE` is treated differently.

```suggestion
        if ActiveRecord::VERSION::STRING >= '6.1'
          index, _, _ = add_index_options(table_name, column_name, **options)
          index_name = index.name
          index_type = index.type
          index_columns = index.columns.first.to_s
          index_options = index.column_options.compact_blank.presence
        else
          index_name, index_type, index_columns, index_options = add_index_options(table_name, column_name, options)
        end
        execute "ALTER TABLE #{quote_table_name(table_name)} ADD #{index_type} INDEX #{quote_column_name(index_name)} (#{index_columns})#{index_options}" # rubocop:disable Metrics/LineLength
      end
```

File: departure.gemspec [hunk 3]
```diff
@@ -30,6 +30,6 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'rake', '~> 10.0'
   spec.add_development_dependency 'rspec', '~> 3.4', '>= 3.4.0'
   spec.add_development_dependency 'rspec-its', '~> 1.2'
-  spec.add_development_dependency 'byebug', '~> 8.2', '>= 8.2.1'
+  spec.add_development_dependency 'pry-byebug'
```
Comment (vivekbisen) [inline departure.gemspec:33]: `byebug` doesn't have colors so I switched it to this. I can revert this change.

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -37,6 +37,7 @@
 
 RSpec.configure do |config|
   config.include TableMethods
+  config.filter_run_when_matching :focus
```
Comment (vivekbisen) [inline spec/spec_helper.rb:40]: [This](https://relishapp.com/rspec/rspec-core/v/3-10/docs/filtering/filter-run-when-matching) allows us to focus on `describe/context/it`s. I can revert this change too.

**Comment by vivekbisen** [comment]
@wyhaines It's ready for review. LMK if you have any comments on this.

**Review by chrismaximin — APPROVED** [review]

**Comment by wyhaines** [comment]
This last week has been insane with the new job, so I apologize for being slow on this.  Looking, but I'll likely merge it in just a couple minutes.

**Pull request merged**